src/app/editor/components/node-editor/node-editor.component.ts

```diff
@@ -154,38 +158,80 @@ export class NodeEditorComponent implements OnInit, OnDestroy {
         if (this.formGenerator.isDirty) {
 
             this.isSaving = true;
-            if (!this.node.uuid) {
-                const parentNode = this.entities.getNode(this.node.parentNode.uuid, { language : this.node.language });
-                this.editorEffects.saveNewNode(parentNode.project.name, this.node)
-                    .then(node => {
-                        this.isSaving = false;
-
-                        if (node) {
-                            this.formGenerator.setPristine(node);
-                            this.listEffects.loadChildren(parentNode.project.name, parentNode.uuid, node.language);
-
-                            if (navigateOnSave) {
-                                this.navigationService.detail(parentNode.project.name, node.uuid, node.language).navigate();
-                            }
+
+            const numFields = Object.keys(getMeshNodeBinaryFields(this.node)).length;
+
+            if (numFields > 0) {
+                this.modalService.fromComponent(ProgressbarModalComponent,
+                                                {
+                                                    closeOnOverlayClick: false,
+                                                    closeOnEscape: false
+                                                },
+                                                {
+                                                    translateToPlural: numFields > 1
+                                                })
+                .then(modal => {
+                    modal.open();
+                    this.saveNodeWithModal(navigateOnSave, modal);
```

Comment:
Refactor this so that saveNodeWithModal returns a promise. Rename that method b/c now has nothing to do with modal.